src/nvim/memline.c

```diff
@@ -2413,23 +2357,25 @@ static int ml_delete_int(buf_T *buf, linenr_T lnum, bool message)
   // find the data block containing the line
   // This also fills the stack with the blocks from the root to the data block
   // This also releases any locked block.
-  mfp = buf->b_ml.ml_mfp;
+  memfile_T *mfp = buf->b_ml.ml_mfp;
   if (mfp == NULL) {
     return FAIL;
   }
 
+  bhdr_T *hp;
   if ((hp = ml_find_line(buf, lnum, ML_DELETE)) == NULL) {
     return FAIL;
   }
 
-  dp = hp->bh_data;
+  DATA_BL *dp = hp->bh_data;
   // compute line count before the delete
```

Comment:
The deleted comment can be moved here and it does make code clearer
```suggestion
  // compute line count (number of entries in block) before the delete
```